src/integration-tests/launchWithEnvironment.spec.ts

```diff
@@ -0,0 +1,345 @@
+            VARTEST2: 'TEST2_SOMEDIFFERENT_VALUE',
+            VARTEST3: 'TEST3_SOMEDIFFERENT_VALUE',
+            VARTEST4: 'TEST4_SOMEDIFFERENT_VALUE',
+        };
+
+        const { APP_PROC, GDB_PROC } = await runForEnvironmentTest(
+            debugTargetAdapter,
+            this.test,
+            environment,
+            targetEnvironment
+        );
+
+        expect(APP_PROC.ENV_VARTEST1).to.equals('TEST1_SOMEDIFFERENT_VALUE');
+        expect(APP_PROC.ENV_VARTEST2).to.equals('TEST2_SOMEDIFFERENT_VALUE');
+        expect(APP_PROC.ENV_VARTEST3).to.equals('TEST3_SOMEDIFFERENT_VALUE');
+        expect(APP_PROC.ENV_VARTEST4).to.equals('TEST4_SOMEDIFFERENT_VALUE');
+
+        expect(GDB_PROC.SHOW_VARTEST1).to.equals('VARTEST1 = TEST1');
+        expect(GDB_PROC.SHOW_VARTEST2).to.equals('VARTEST2 = TEST2');
+        expect(GDB_PROC.SHOW_VARTEST3).to.equals(
+            'Environment variable "VARTEST3" not defined.'
+        );
+        expect(GDB_PROC.SHOW_VARTEST4).to.equals(
+            'Environment variable "VARTEST4" not defined.'
+        );
+    });
+
+    it('unsets when target environment variables sets null with debugTargetAdapter', async function () {
+        if (hardwareBreakpoint) {
+            this.skip();
+        }
+        const environment = {
+            VARTEST1: 'TEST1',
+            VARTEST2: 'TEST2',
+            VARTEST3: 'TEST3',
+            VARTEST4: 'TEST4',
+        };
+        const targetEnvironment = {
+            VARTEST1: 'TEST1_SOMEDIFFERENT_VALUE',
+            VARTEST2: 'TEST2_SOMEDIFFERENT_VALUE',
+            VARTEST3: null,
+            VARTEST4: null,
+        };
+
+        const { APP_PROC, GDB_PROC } = await runForEnvironmentTest(
+            debugTargetAdapter,
+            this.test,
+            environment,
+            targetEnvironment
+        );
+
+        expect(APP_PROC.ENV_VARTEST1).to.equals('TEST1_SOMEDIFFERENT_VALUE');
+        expect(APP_PROC.ENV_VARTEST2).to.equals('TEST2_SOMEDIFFERENT_VALUE');
+        expect(APP_PROC.ENV_VARTEST3).to.equals(null);
+        expect(APP_PROC.ENV_VARTEST4).to.equals(null);
+
+        expect(GDB_PROC.SHOW_VARTEST1).to.equals('VARTEST1 = TEST1');
+        expect(GDB_PROC.SHOW_VARTEST2).to.equals('VARTEST2 = TEST2');
+        expect(GDB_PROC.SHOW_VARTEST3).to.equals('VARTEST3 = TEST3');
+        expect(GDB_PROC.SHOW_VARTEST4).to.equals('VARTEST4 = TEST4');
+    });
+
+    it('ensures that path is not null', async function () {
+        if (hardwareBreakpoint) {
+            this.skip();
+        }
+        const { APP_PROC } = await runForEnvironmentTest(undefined, this.test);
+
+        expect(APP_PROC.ENV_PATH).not.to.equals(null);
+    });
+
+    it('ensures that new entries could be injected to path', async function () {
+        if (hardwareBreakpoint) {
+            this.skip();
+        }
+        const pathToAppend = __dirname;
+        const currentPathValue = process.env.PATH || process.env.Path;
```

Comment:
This is probably fine, but path can be anycase (i.e. `process.env.path`). In practice it almost never is anything other than PATH or Path on the test machines.

Reply:
Not sure to update this one, because I am returning checking the same casing in C code too. "ensures that path is not null" test is implicitly controlling this. I can still update this if you think it could cause a problem in the future.

Reply:
I just tested the assumption here that KEY in `process.env.KEY` is case sensitive, it is not case sensitive on Windows, which matches the [documentation](https://nodejs.org/api/process.html#processenv):

> On Windows operating systems, environment variables are case-insensitive.


```suggestion
        const currentPathValue = process.env.PATH;
```